bin/tf

```diff
@@ -225,6 +227,39 @@ def process_arguments(args: List[str]) -> Tuple[str, str, List[str], bool]:
         sys.exit(0)
 
 
+def download_plugins(plugin_paths: Dict[str, str]):
+    for name, path in plugin_paths.items():
+        home = os.path.expanduser("~")
+        file_path = os.path.join(home, '.terraform.d/plugins', name)
+        etag_path = '%s.%s' % (file_path, 'etag')
+
+        # get the etag if it exists
+        headers = {}
+        if os.path.isfile(etag_path):
+            with(open(etag_path, 'r')) as etag_file:
+                headers['If-None-Match'] = etag_file.read()
+
+        # get the plugin. send etag header to avoid downloading file if we already have the same version
+        response = requests.get(path, headers=headers)
+        response.raise_for_status()
```

Comment:
Should this still be here?

Reply:
This will cause `requests` to raise an exception if it cant download the plugin so we don't need to check the status and raise an error ourselves